wagtail/wagtailcore/models.py

```diff
@@ -763,13 +763,17 @@ def move(self, target, pos=None):
         logger.info("Page moved: \"%s\" id=%d path=%s", self.title, self.id, new_url_path)
 
     def copy(self, recursive=False, to=None, update_attrs=None, copy_revisions=True, keep_live=True, user=None):
-        # Make a copy
-        page_copy = Page.objects.get(id=self.id).specific
-        page_copy.pk = None
-        page_copy.id = None
-        page_copy.depth = None
-        page_copy.numchild = 0
-        page_copy.path = None
+        # Fill dict with self.specific values
+        exclude_fields = ['id', 'path', 'depth', 'numchild', 'url_path', 'path']
+        specific_self = self.specific
+        specific_dict = {}
+
+        for field in specific_self._meta.fields:
```

Comment:
This should use the new Django 1.8 meta API by default (https://docs.djangoproject.com/en/1.8/ref/models/meta/#retrieving-all-field-instances-of-a-model)

If that doesn't work well on Django 1.7, feel free to fallback to what you already have now with a comment